### ISSUE TITLE
Odd behavior from readButtons() on type 1 module

### ISSUE THREAD

**Issue by marclefevre** [gavinlyonsrepo/TM1638plus#24]
I'm using your library to access a type one module.  It has 8 buttons but the data I get back from readButtons() is odd.  When I look at the function in TM1638plus.cpp I see that it seems to be only shifting out 4 bits from the device:

`	for (uint8_t i = 0; i < 4; i++)
	{

		if  (_HIGH_FREQ == false)
			v = shiftIn(_DATA_IO, _CLOCK_IO, LSBFIRST) << i;
		else
			v = HighFreqshiftin(_DATA_IO, _CLOCK_IO, LSBFIRST) << i;

		buttons |= v;
	}
`
I tried changing the 4 in the for loop to an 8 and that caused most of the other functions to stop working correctly.  My question is, is the code in readButtons() correct?

I'm using library version 2.0.0.

**Comment by gavinlyonsrepo** [comment]
Hi 

Define Odd behaviour please?
What Micro-controller are you using ?

Yes code is correct. We read 4 bytes back from device, the 'i' variable is used to position the relevant returned bits in each byte into a single byte which is then returned as variable 'buttons'. 

See image  for
![tm1638canvas3](https://github.com/user-attachments/assets/427ceb59-0599-4348-a734-5a87d281553e)
 more details.

Regards

**Comment by marclefevre** [comment]
OK.  I get it now.  I still have a problem to solve, but I know that the
issue is with my code and not yours.

Thanks.

**Comment by gavinlyonsrepo** [comment]
What Micro-controller are you using ?

**Comment by marclefevre** [comment]
ESP8266-12

**Comment by gavinlyonsrepo** [comment]
There is a known issue with reading the buttons on 'High' frequency micro-controllers like ESP 32 see note 2  in the readme Notes section.  "High frequency micro-controllers."

You can try 
1. Running the ESp2866 at a lower CPU frequency (this is a setting in arduino IDE Tools) 
2. Set the High frequency setting in sketch to true 

` bool high_freq = true; // default false, If using a high freq CPU > ~100 MHZ set to true. 
// Constructor object (GPIO STB , GPIO CLOCK , GPIO DIO, use high freq MCU)
TM1638plus tm(STROBE_TM, CLOCK_TM , DIO_TM, high_freq);`

**Comment by marclefevre** [comment]
I do have the high_freq argument to the constructor set to true.  Is there
still an issue for the 8266?

**Comment by gavinlyonsrepo** [comment]
I have not seen any problem with ESP8266 and the read buttons function.
Only ESP32 at 240MHZ and Teensy at 396Mhz. 
I was just suggesting it in cause it was the case for you.

I retested there now to be sure. I did notice the 'Test12' Scroll test does not work for ESP8266 in file TM1638plus_TEST_Model1.ino it restarts the ESP8266. Although that's another issue.

Are you using the code from example file or your own code?
